src/views/Notifications/Home/_components/_utils.tsx

```diff
@@ -0,0 +1,211 @@
+  GetNotifications_currentUser_notifications_notifications_activity_metaData_TaggedInPlanDiscussionActivityMeta as TaggedInDiscussionActivityType
+} from 'gql/gen/types/GetNotifications';
+import i18next from 'i18next';
+
+// Type guard to check union type
+export const isTaggedInDiscussion = (
+  data:
+    | TaggedInDiscussionReplyActivityType
+    | TaggedInDiscussionActivityType
+    | DailyDigestCompleteActivityType
+): data is TaggedInDiscussionActivityType => {
+  /* eslint no-underscore-dangle: 0 */
+  return data.__typename === 'TaggedInPlanDiscussionActivityMeta';
+};
+
+export const isTaggedInDiscussionReply = (
+  data:
+    | TaggedInDiscussionReplyActivityType
+    | TaggedInDiscussionActivityType
+    | DailyDigestCompleteActivityType
+): data is TaggedInDiscussionReplyActivityType => {
+  /* eslint no-underscore-dangle: 0 */
+  return data.__typename === 'TaggedInDiscussionReplyActivityMeta';
+};
+
+export const isDailyDigest = (
+  data:
+    | TaggedInDiscussionReplyActivityType
+    | TaggedInDiscussionActivityType
+    | DailyDigestCompleteActivityType
+): data is DailyDigestCompleteActivityType => {
+  /* eslint no-underscore-dangle: 0 */
+  return data.__typename === 'DailyDigestCompleteActivityMeta';
+};
+
+export const activityText = (data: MetaDataType) => {
+  if (isTaggedInDiscussion(data)) {
+    return (
+      <Trans
+        i18nKey="notifications:index.activityType.taggedInDiscussion.text"
+        values={{ modelName: data.modelPlan.modelName }}
+      />
+    );
+  }
+  if (isTaggedInDiscussionReply(data)) {
+    return (
+      <Trans
+        i18nKey="notifications:index.activityType.taggedInDiscussionReply.text"
+        values={{ modelName: data.modelPlan.modelName }}
+      />
+    );
+  }
+  if (isDailyDigest(data)) {
+    return (
+      <Trans i18nKey="notifications:index.activityType.dailyDigestComplete.text" />
+    );
+  }
+  return '';
+};
+
+export const ActivityCTA = ({
+  data,
+  isExpanded
+}: {
+  data: MetaDataType;
+  isExpanded: boolean;
+}) => {
+  if (isTaggedInDiscussion(data)) {
+    return (
+      <>
+        <Trans i18nKey="notifications:index.activityType.taggedInDiscussion.cta" />
+        <Icon.ArrowForward className="margin-left-1" aria-hidden />
+      </>
+    );
+  }
+  if (isTaggedInDiscussionReply(data)) {
+    return (
+      <>
+        <Trans i18nKey="notifications:index.activityType.taggedInDiscussionReply.cta" />
+        <Icon.ArrowForward className="margin-left-1" aria-hidden />
+      </>
+    );
+  }
+  if (isDailyDigest(data)) {
+    return (
+      <>
+        <Trans i18nKey="notifications:index.activityType.dailyDigestComplete.cta" />
+        {isExpanded ? (
+          <Icon.ExpandLess className="margin-left-1" aria-hidden />
+        ) : (
+          <Icon.ExpandMore className="margin-left-1" aria-hidden />
+        )}
+      </>
+    );
+  }
+  return <></>;
+};
+export const TranslateStatusChange = ({
+  status
+}: {
+  status: string | null;
+}) => {
```

Comment:
I think we could remove this util function/component if we just format the translations to use the keys of the status.  Also this should be using the a gql type `ModelStatus` I believe.  You can use TS in the translation with `ModelStatus`

Ex: Translation file
```
statusChanges: {
    PLAN_DRAFT: 'Plan draft',
    PLAN_COMPLETE: 'Plan complete'
}
```